drivers/hmc7044/hmc7044.c

```diff
@@ -235,6 +236,33 @@ static int hmc7044_write(struct hmc7044_dev *dev,
 	return spi_write_and_read(dev->spi_desc, buf, ARRAY_SIZE(buf));
 }
 
+/**
+ * SPI register read from device.
+ * @param dev - The device structure.
+ * @param reg - The register address.
+ * @param val - The register data.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int hmc7044_read(struct hmc7044_dev *dev,
+			 uint16_t reg,
+			 uint8_t *val)
+{
+	uint8_t buf[3];
+	uint16_t cmd;
+	int ret;
+
+	cmd = HMC7044_READ | HMC7044_CNT(1) | HMC7044_ADDR(reg);
+	buf[0] = cmd >> 8;
+	buf[1] = cmd & 0xFF;
+	buf[2] = 0;
+
+	ret = spi_write_and_read(dev->spi_desc, buf, ARRAY_SIZE(buf));
+
+	*val = buf[2];
+
+	return ret;
```

Comment:
Curious: does it make sense to update `val ` if the `spi_write_and_read()` has failed?
```
ret = spi_write_and_read(dev->spi_desc, buf, ARRAY_SIZE(buf));
if (ret < 0)
        return ret;

*val = buf[2];

return 0;
```